onnxruntime/core/providers/cpu/activation/activations.h

```diff
@@ -98,7 +98,7 @@ struct Softplus : public ElementWiseRangedTransform<T> {
     T* output_ptr = this->output + first;
     ConstEigenVectorArrayMap<T> xm(this->input + first, len);
     EigenVectorArrayMap<T> ym(output_ptr, len);
-    ym = (xm > 0).select(xm + ((-xm).exp() + 1.0f).log(), ((xm).exp() + 1.0f).log());
+    ym = (xm > 0).select(xm + ((-xm).exp()).log1p(), ((xm).exp()).log1p());
```

Comment:
There is a bit concern here, since Eigen log use SSE2 and AVX, while log1p use std::log1p, this may cause the perf loss on certain platform.

See, https://eigen.tuxfamily.org/dox/group__CoeffwiseMathFunctions.html

Reply:
@yufenglee is there anything in mlas that could be used here?

Reply:
MLAS has MlasComputeExp and MlasComputeLogistic, but doesn't have one for Softplus. If we compute Softplus with MlasComputeExp and MlasComputeLogistic, it will go though memory multiple times. 

Looks like this activation is not used commonly. If it is, we can add optimization for it.

Reply:
AFAIK Softmax is typically used as a post-processing step and often done outside of the model or as a final step inside the model. As such perf may not be  critical.

Reply:
This is for Softplus, not Softmax

Reply:
Ahh sorry - I have never seen a model that uses Softplus.

Correctness has to come first. If there's a perf issue that affects a production model we can look at adding something to MLAS in the future. That said, from some very brief research it seems like Relu is generally preferred over Softplus, so any perf issue could potentially be addressed by switching to that.